### PR TITLE
Bug/repair warnings for your mission comp

### PR DESCRIPTION
**Pull request by OniWarrior**
In this ticket, I resolved several defined but never used warnings:

./src/components/pages/Gamemode/YourMissionComp.js
Line 1:27: 'useEffect' is defined but never used
Line 5:20: 'Button' is defined but never used
Line 7:8: 'UploaderComp' is defined but never used
Line 8:10: 'tasks' is defined but never used
Line 9:10: 'Gamemode' is defined but never used
Line 10:10: 'Link' is defined but never used
Line 10:16: 'Route' is defined but never used
Line 11:10: 'render' is defined but never used
Line 14:11: 'push' is assigned a value but never used
Line 14:17: 'location' is assigned a value but never used
Line 15:15: 'setsRwd' is assigned a value but never used

In the YourMissionComp.js file, there were several imports that were imported that were never used.
From line 1-11, I deleted the imports that were triggering the warnings. Note: If we need a particular import in the future, we can import what we need when we are ready to implement. I deleted the push and location consts that were not being used on line 14.

On line 15 there was a state variable and its update function that was declared. The state variable "rwd" was being used throughout the codebase but the update function, setsRwd, was not being used. The individual who coded this component was trying to directly assign values to the properties of the state variable rather than use the update function to update those values properly. It didn't make sense when I saw this. In any case, I deleted the direct assignments on #20,#22,#24,#34#,36,and #38. I then made a call to the update function, setsRwd, and passed the values from props to the individual properties to update the state variable. I made a call on #25-29 and #39-43.

I understood what their intention was but the way the individual implemented wasn't correct. With setsRwd implemented, the warning was resolved.

--------------------------------------------------------------------------------------------------
PR submission vid: https://drive.google.com/file/d/10WIrJoXHl_cbZWxeeCtOJYmq7BNXTFVx/view?usp=sharing
Trello card: https://trello.com/c/0PLOlsOR/603-defined-but-never-used-f

### CODE REVIEW

**Review by rhiannonstanford — APPROVED** [review]
All changes have been reviewed in previous PR submission.  Excellent job setting state properly and getting this component to function.  Component / page was tested and is working as expected.  Page renders text and modal conditionally as intended.

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect } from 'react';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:1]: great job on removing this unneeded hook

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { useState } from 'react';
 import UploadDocs from '../../common/UploadDocs';
 import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
-import { Row, Col, Button } from 'antd';
+import { Row, Col } from 'antd';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:5]: No button found within the code. Great catch!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 3]
```diff
@@ -4,7 +4,6 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import UploaderComp from './UploaderComp';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:7]: UploaderComp was not used great catch!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 4]
```diff
@@ -4,7 +4,6 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import { tasks } from '../../../state/actions';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:8]: tasks action was not used! great catch

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 5]
```diff
@@ -4,7 +4,6 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import { Gamemode } from './index';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:9]: gamemode was not used. another great catch

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 6]
```diff
@@ -4,7 +4,6 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import { Link, Route, useHistory } from 'react-router-dom';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:10]: React router was not used here. Great catch

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 7]
```diff
@@ -4,7 +4,6 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import { render } from 'react-dom';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:11]: Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 8]
```diff
@@ -6,7 +6,6 @@ import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
 import Thrashbar from './Thrashbar';
 const YourMissionComp = ({ ...props }) => {
-  const { push, location } = useHistory();
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:14]: useHistory was not used here

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 9]
```diff
@@ -15,7 +15,7 @@ const YourMissionComp = ({ ...props }) => {
   const fxd = () => {
     if (props.child.gamemode.write) {
       props.child.gamemode.write = true;
-      rwd.write = true;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:24]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 10]
```diff
@@ -17,7 +17,7 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.write = true;
 
       props.child.gamemode.draw = false;
-      rwd.draw = false;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:26]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 11]
```diff
@@ -19,7 +19,6 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.draw = false;
 
       props.child.gamemode.read = true;
-      rwd.read = true;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:28]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 12]
```diff
@@ -19,6 +19,11 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.draw = false;
 
       props.child.gamemode.read = true;
+      setsRwd({
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:22]: wonderful implementation of props

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 13]
```diff
@@ -28,7 +28,7 @@ const YourMissionComp = ({ ...props }) => {
 
     if (props.child.gamemode.draw) {
       props.child.gamemode.write = false;
-      rwd.write = false;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:33]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 14]
```diff
@@ -30,7 +30,7 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.write = false;
 
       props.child.gamemode.draw = true;
-      rwd.draw = true;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:35]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 15]
```diff
@@ -32,7 +32,6 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.draw = true;
 
       props.child.gamemode.read = true;
-      rwd.read = true;
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:37]: Good job removing this. Props was not properly used here. Good job!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 16]
```diff
@@ -32,6 +32,11 @@ const YourMissionComp = ({ ...props }) => {
       props.child.gamemode.draw = true;
 
       props.child.gamemode.read = true;
+      setsRwd({
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:35]: Great work!

**Review by just-kato — DISMISSED** [review]

File: src/components/pages/Gamemode/YourMissionComp.js (hunk 8 above)
Comment (OniWarrior) [reply src/components/pages/Gamemode/YourMissionComp.js:14]: No, I saw no indication that it was going to be used.

**Review by Yonathan-Admasu728 — APPROVED** [review]
Clear and clean fix. all 16 commits, shows how carefully you are making those changes as you go through the code line by line. Excellent Job!

**Pull request merged**